hail/python/hailtop/httpx.py

```diff
@@ -83,9 +83,34 @@ async def __aexit__(
 
 
 class ClientSession:
-    def __init__(self, *args, **kwargs):
-        self.raise_for_status = kwargs.pop('raise_for_status', False)
-        self.client_session = aiohttp.ClientSession(*args, raise_for_status=False, **kwargs)
+    def __init__(self,
+                 *args,
+                 raise_for_status: bool = True,
+                 timeout: Union[aiohttp.ClientTimeout, float] = None,
+                 **kwargs):
+        location = get_deploy_config().location()
+        if location == 'external':
+            tls = external_client_ssl_context()
+        elif location == 'k8s':
+            tls = internal_client_ssl_context()
+        else:
+            assert location in ('gce', 'azure')
+            # no encryption on the internal gateway
+            tls = external_client_ssl_context()
+
+        assert 'connector' not in kwargs
+        kwargs['connector'] = aiohttp.TCPConnector(ssl=tls)
```

Comment:
Nit: a little awkward here putting `connector` back in `kwargs` but not `timeout`. Can we leave `kwargs` as it is and pass `connector` explicitly to `ClientSession`?